### PR TITLE
KAFKA-3424: Add CORS support to Connect REST API

### PR DESCRIPTION
**Pull request by ewencp**

### CODE REVIEW

**Comment by ewencp** [comment]
@gwenshap @Ishiihara @hachikuji Need a review on this -- it's pretty straightforward though. We have similar code in Confluent's rest-utils, but here it is even properly tested!

File: connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestServerTest.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        workerProps.put(WorkerConfig.ACCESS_CONTROL_ALLOW_ORIGIN_CONFIG, "*");
+        WorkerConfig workerConfig = new StandaloneConfig(workerProps);
+        server = new RestServer(workerConfig);
+        server.start(herder);
+
+        Response response = request("/connectors")
+                .header("Referer", "http://bar.com/page")
+                .header("Origin", "http://bar.com")
+                .get();
+        assertEquals(200, response.getStatus());
```
Comment (gwenshap) [inline connect/runtime/src/test/java/org/apache/kafka/connect/runtime/rest/RestServerTest.java:107]: Do we also want to test the negative? That requests are rejected when the flag is not set?

**Comment by ewencp** [comment]
@gwenshap Updated to test when we don't have CORS enabled -- it won't fail the request, it's just going to not return the header.

**Comment by gwenshap** [comment]
LGTM

**Pull request closed without merging**